base/MAPL_ISO8601_Time.F90

```diff
@@ -0,0 +1,720 @@
+      if(factor /= 0) then ! To avoid divide by 0
+          is_factor = mod(dividend, factor)==0
+      else
+          is_factor = .false.
+      endif
+   end function is_factor
+
+   ! Return true if factor does not divide dividend evenly, false otherwise
+   ! see is_factor
+   pure logical function is_not_factor(dividend, factor)
+      integer, intent(in) :: dividend
+      integer, intent(in) :: factor
+      is_not_factor = .not. is_factor(dividend, factor)
+   end function is_not_factor
+
+   pure logical function is_between(lower, upper, n) 
+      integer, intent(in) :: lower
+      integer, intent(in) :: upper
+      integer, intent(in) :: n
+      is_between = n > lower .and. n < upper
+   end function is_between
+
+   ! Return true if y is a leap year, false otherwise
+   pure logical function is_leap_year(y)
+      integer, intent(in) :: y
+      ! Leap years are years divisible by 400 or (years divisible by 4 and not divisible by 100)
+      is_leap_year = is_factor(y,400) .or. ( is_factor(y, 4) .and. is_not_factor(y, 100) )
+   end function is_leap_year
+
+   ! Return the last day numbers of each month based on the year
+   pure function get_month_ends(y) result(month_ends)
+      integer, intent(in) :: y
+      integer, dimension(NUM_MONTHS) :: month_ends
+      ! last day numbers of months for leap years
+      integer, dimension(NUM_MONTHS) :: MONTH_END_LEAP
+      ! last day numbers of months for regular years
+      integer, dimension(NUM_MONTHS) :: MONTH_END 
+      MONTH_END = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+      MONTH_END_LEAP = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+
+      if(is_leap_year(y)) then
+         month_ends = MONTH_END_LEAP 
+      else
+         month_ends = MONTH_END
+      endif
+   end function get_month_ends
+
+   ! Return the last day number of month m in year y
+   pure integer function get_month_end(y, m)
+      integer, intent(in) :: y
+      integer, intent(in) :: m
+      integer, dimension(NUM_MONTHS) :: month_ends
+
+      month_ends = get_month_ends(y)    
+      get_month_end = month_ends(m)
+
+   end function get_month_end
+
+   pure logical function is_valid_year(y)
+      integer, intent(in) :: y
+      is_valid_year = is_between(LB_YEAR, UB_YEAR, y)
+   end function is_valid_year 
+
+   pure logical function is_valid_month(m)
+      integer, intent(in) :: m
+      is_valid_month = is_between(LB_MONTH, UB_MONTH, m)
+   end function is_valid_month
+
+!   pure logical function is_valid_day_ints(y, m, d)
+!      integer, intent(in) :: y
+!      integer, intent(in) :: m
+!      integer, intent(in) :: d
+!      integer :: month_end
+!
+!      if(is_valid_year(y) .and. is_valid_month(m)) then
+!         month_end = get_month_end(y, m)
+!         is_valid_day_ints = is_between(LB_DAY, month_end+1, d) 
+!      else
+!         is_valid_day_ints = .FALSE.
+!      endif
+!   end function is_valid_day_ints
```

Comment:
Reminder to delete if not used.  (This comment is probably more visible than the comment characters - just trying to help.)